src/granite_orm/fields.cr

```diff
@@ -103,93 +103,42 @@ module Granite::ORM::Fields
-              @{{_name.id}} = value.to_i32
-            {% elsif type.id == Int64.id %}
-              @{{_name.id}} = value.to_i64
-            {% elsif type.id == Float32.id %}
-              @{{_name.id}} = value.to_f32{0.0}
-            {% elsif type.id == Float64.id %}
-              @{{_name.id}} = value.to_f64{0.0}
-            {% elsif type.id == Bool.id %}
-              @{{_name.id}} = ["1", "yes", "true", true].includes?(value)
-            {% elsif type.id == Time.id %}
-              if value.is_a?(Time)
-                @{{_name.id}} = value
-              elsif value.to_s =~ TIME_FORMAT_REGEX
-                @{{_name.id}} = Time.parse(value.to_s, "%F %X")
-              end
-            {% else %}
-              @{{_name.id}} = value.to_s
-            {% end %}
-          {% end %}
-        end
-      else
-        case name.to_s
-          {% for _name, type in FIELDS %}
-          when "{{_name.id}}"
-            @{{_name.id}} = nil
-          {% end %}
-        end
-      end
-    end
-
-    # Casts params and sets fields for JSON::Type
-    private def cast_json_to_field(name, value : JSON::Type | Int32 | Float32 | Time)
-       if !value.nil?
-        case name.to_s
-          {% for _name, type in FIELDS %}
-          when "{{_name.id}}"
-            {% if type.id == Int32.id %}
-              @{{_name.id}} = value.as(Int64).to_i32
-            {% elsif type.id == Int64.id %}
-              @{{_name.id}} = value.as(Int64)
-            {% elsif type.id == Float32.id %}
-              @{{_name.id}} = value.as(Float64).to_f32
-            {% elsif type.id == Float64.id %}
-              @{{_name.id}} = value.as(Float64)
-            {% elsif type.id == Bool.id %}
-              @{{_name.id}} = value.as(Bool)
-            {% elsif type.id == Time.id %}
-              if value.is_a?(Time)
-                @{{_name.id}} = value
-              elsif value.to_s =~ TIME_FORMAT_REGEX
-                @{{_name.id}} = Time.parse(value.to_s, "%F %X")
-              end
-            {% else %}
-              @{{_name.id}} = value.to_s
-            {% end %}
-          {% end %}
-        end
-      else
-        case name.to_s
-          {% for _name, type in FIELDS %}
-          when "{{_name.id}}"
-            @{{_name.id}} = nil
+    # Casts params and sets fields
+    private def cast_to_field(name, value : Type)
+      case name.to_s
+        {% for _name, type in FIELDS %}
+        when "{{_name.id}}"
+          return @{{_name.id}} = nil if value.nil?
+          {% if type.id == Int32.id %}
+            @{{_name.id}} = value.is_a?(String) ? value.to_i32 : value.is_a?(Int64) ? value.to_s.to_i32 : value.as(Int32)
+          {% elsif type.id == Int64.id %}
+            @{{_name.id}} = value.is_a?(String) ? value.to_i64 : value.as(Int64)
+          {% elsif type.id == Float32.id %}
+            @{{_name.id}} = value.is_a?(String) ? value.to_f32 : value.is_a?(Float64) ? value.to_s.to_f32 : value.as(Float32)
```

Comment:
Parenthesis or some sort of mapping method could make this logic readable...

value.is_a?(String) ? value.to_f32 : (value.is_a?(Float64) ? value.to_s.to_f32 : value.as(Float32))